include/aero/actuator/ActuatorGenericTurbineSearchFunctor.h

```diff
@@ -0,0 +1,147 @@
+// Copyright 2017 National Technology & Engineering Solutions of Sandia, LLC
+// (NTESS), National Renewable Energy Laboratory, University of Texas Austin,
+// Northwest Research Associates. Under the terms of Contract DE-NA0003525
+// with NTESS, the U.S. Government retains certain rights in this software.
+//
+// This software is released under the BSD 3-clause license. See LICENSE file
+// for more details.
+//
+// Copyright 2017 National Technology & Engineering Solutions of Sandia, LLC
+// (NTESS), National Renewable Energy Laboratory, University of Texas Austin,
+// Northwest Research Associates. Under the terms of Contract DE-NA0003525
+// with NTESS, the U.S. Government retains certain rights in this software.
+//
+// This software is released under the BSD 3-clause license. See LICENSE file
+// for more details.
+//
+#ifndef ACTUATORGENERICTURBINESEARCHFUNCTOR_H_
+#define ACTUATORGENERICTURBINESEARCHFUNCTOR_H_
+
+#include <aero/actuator/UtilitiesActuator.h>
+#include <aero/actuator/ActuatorTypes.h>
+#include <stk_mesh/base/BulkData.hpp>
+#include <FieldTypeDef.h>
+
+namespace sierra {
+namespace nalu {
+
+template <typename ActuatorBulk, typename functor>
+  //coarse search actuatorbulk.c L96
+struct GenericLoopOverCoarseTurbineSearchResults
+{
+  using execution_space = ActuatorFixedExecutionSpace;
+
+  // ctor if functor only requires actBulk for constructor
+  GenericLoopOverCoarseTurbineSearchResults(
+    ActuatorBulk& actBulk, stk::mesh::BulkData& stkBulk)
+    : actBulk_(actBulk),
+      stkBulk_(stkBulk),
+      coordinates_(stkBulk_.mesh_meta_data().template get_field<double>(
+        stk::topology::NODE_RANK, "coordinates")),
+      actuatorSource_(stkBulk_.mesh_meta_data().template get_field<double>(
+        stk::topology::NODE_RANK, "actuator_source")),
+      dualNodalVolume_(stkBulk_.mesh_meta_data().template get_field<double>(
+        stk::topology::NODE_RANK, "dual_nodal_volume")),
+      innerLoopFunctor_(actBulk)
+  {
+    actBulk_.coarseSearchElemIds_.sync_host();
+    actBulk_.coarseSearchPointIds_.sync_host();
+    innerLoopFunctor_.preloop();
+  }
+
+  // ctor for functor constructor taking multiple args
+  GenericLoopOverCoarseTurbineSearchResults(
+    ActuatorBulk& actBulk,
+    stk::mesh::BulkData& stkBulk,
+    functor innerLoopFunctor)
+    : actBulk_(actBulk),
+      stkBulk_(stkBulk),
+      coordinates_(stkBulk_.mesh_meta_data().template get_field<double>(
+        stk::topology::NODE_RANK, "coordinates")),
+      actuatorSource_(stkBulk_.mesh_meta_data().template get_field<double>(
+        stk::topology::NODE_RANK, "actuator_source")),
+      dualNodalVolume_(stkBulk_.mesh_meta_data().template get_field<double>(
+        stk::topology::NODE_RANK, "dual_nodal_volume")),
+      innerLoopFunctor_(innerLoopFunctor)
+  {
+    actBulk_.coarseSearchElemIds_.sync_host();
+    actBulk_.coarseSearchPointIds_.sync_host();
+    innerLoopFunctor_.preloop();
```

Comment:
`innerLoopExtent = actBulk.singlePointCoarseSearch ? 1 : /*actuatorBulk points extent*/;`
Then add in the inner loop inside the turbine